src/EnergyPlus/DataZoneEquipment.cc

```diff
@@ -382,180 +379,171 @@ void GetZoneEquipmentData(EnergyPlusData &state)
         // for each equipment list
 
         CurrentModuleObject = "ZoneHVAC:EquipmentList";
+        auto &ip = state.dataInputProcessing->inputProcessor;
 
-        ZoneEquipListNum = state.dataInputProcessing->inputProcessor->getObjectItemNum(
-            state, CurrentModuleObject, state.dataZoneEquip->ZoneEquipConfig(ControlledZoneNum).EquipListName);
-        if (ZoneEquipListNum > 0) {
+        ZoneEquipListNum = ip->getObjectItemNum(state, CurrentModuleObject, state.dataZoneEquip->ZoneEquipConfig(ControlledZoneNum).EquipListName);
+
+        if (ZoneEquipListNum <= 0) {
+            ShowSevereError(state,
+                            std::string{RoutineName} + CurrentModuleObject +
+                                " not found = " + state.dataZoneEquip->ZoneEquipConfig(ControlledZoneNum).EquipListName);
+            ShowContinueError(
+                state, "In ZoneHVAC:EquipmentConnections object, for Zone = " + state.dataZoneEquip->ZoneEquipConfig(ControlledZoneNum).ZoneName);
+            state.dataZoneEquip->GetZoneEquipmentDataErrorsFound = true;
+        } else {
+            auto const &epListFields =
+                ip->getObjectItem(state, CurrentModuleObject, state.dataZoneEquip->ZoneEquipConfig(ControlledZoneNum).EquipListName);
```

Comment:
Here's the use case for the new json InputProcessor::getObjectItem one.

Reply:
May be better to name this function something else, `GetObjectItemJson`? It's essentially a completely different function than the current version.

Reply:
Renamed to getJSONObjectItem: e3b36fe4a552f0b582794816c0e40871a7961dae